Add support for the Shearwater Swift GPS

mikeller opened the pull request:
<!-- Lines like this one are comments and will not be shown in the final output. -->
<!-- Make sure that you have read the "Contributing" section of the README and also the notes in CodingStyle. -->
<!-- If you are a collaborator, please add labels and assign other collaborators for a review. -->

### Describe the pull request:
<!-- Replace [ ] with [x] to select options. -->
- [ ] Bug fix
- [ ] Functional change
- [x] New feature
- [ ] Code cleanup
- [ ] Build system change
- [ ] Documentation change
- [ ] Language translation

### Pull request long description:
<!-- Describe your pull request in detail. -->
Pro forma pull request to generate test artifacts for the changes in the
libdivecomputer submodule.

### Changes made:
<!-- Enumerate the changes with 1), 2), 3) etc. -->
<!-- Ensure the test cases are updated if needed. -->

### Related issues:
<!-- Reference issues with #<issue-num>. -->
<!-- Write "Fixes #<issue-num" to notify Github that this PR fixes an issue. -->
Implements libdivecomputer/libdivecomputer#53.
Fixes subsurface/libdc#97.

### Additional information:
<!-- Include sample dive log or other relevant information to allow testing the change where feasible. -->

### Documentation change:
<!-- If this PR makes changes to user functionality, then the documentation has to be updated too. -->
<!-- Please, briefly outline here what has changed in terms of the user experience (UX). -->
<!-- If UX changes have been made, a maintainer should apply the 'needs-documentation-change' label. -->

### Mentions:
<!-- Mention users that you want to review your pull request with @<user-name>. Leave empty if not sure. -->

dirkhh reviewed (approved):
why a sane person would want this to be a float and not an integer is beyond me, but whatever. Appoved

dirkhh commented:
I would rather fix the libdc code and match the way we get gps days from the Garmin
Random floats are not a reasonable way to do this. 

This has been an ongoing debate and fundamental disagreement for a decade with many other examples.

mikeller commented:
> why a sane person would want this to be a float and not an integer is beyond me, but whatever. Appoved

If I got a dollar each time I saw floats used for lat / lon in 'commercial grade' software I'd be filthy rich by now...

dirkhh commented:
If I got a cent (US$ or NZ$, doesn't matter), for every time I saw a stupid choice of data type and data transfer protocol or data encapsulation protocol, I could buy myself quite a few nice toys...

mikeller commented:
@dirkhh: Not sure where we want to take this from here?

I can see the following options:
1. extend existing approach in Subsurface to use an Extra Info field with a special name to convey location information, and retrofit this to all dive computers that report location information (for now);
2. leverage the the `DC_FIELD_LOCATION` approach that libdivecomputer has implemented for the dive computers supporting location (except for Garmin), and accept that using floats for lat / lon is a [flawed approach](https://xkcd.com/2170/);
3. leverage `DC_FIELD_LOCATION`, but change it to use a fixed decimal notation for lat / lon.

The way I see it, 1. is not particularly sustainable as this will get us to play catch up whenever a new dive computer has got location support. And I really don't think that 'leverage special names for Extra Info to transmit data' is an approach that should be turned into best practice and used more widely. 2. gets us to swallow the frog of float lat / lons, but at least new dive computers will work out of the box. 3. is most work, but then it will fail fast and get us to update the backend for new dive computers if they support location, in order to make them build in Subsurface.

mikeller commented:
@chaehni Can you please try importing again, with the latest artifacts?

chaehni commented:
Hi @mikeller 
Thank you for the update. I downloaded some of the newest dives to a fresh logbook. Unfortunately, I still can't see any location information.
One small thing I noticed, the linked artifact here in Github is called patch.2 whereas the file contained in the zip is patch.3.
Not sure if that is intended and makes a difference here.

mikeller commented:
Thanks @chaehni.

> Hi @mikeller Thank you for the update. I downloaded some of the newest dives to a fresh logbook. Unfortunately, I still can't see any location information.

My bad, I removed the code that populates the location when I did the final cleanup of my test code before pushing. Fixed now.

> One small thing I noticed, the linked artifact here in Github is called patch.2 whereas the file contained in the zip is patch.3. Not sure if that is intended and makes a difference here.

That's expected. It's a consequence of GitHub running the CICD build on a speculative merge of the pull request content into the target branch - so there's an extra commit in the version that is built.
